dhall-openapi/src/Dhall/Kubernetes/Convert.hs

```diff
@@ -102,63 +109,121 @@ namespacedObjectFromImport _ = Nothing
 toTextLit :: Text -> Expr
 toTextLit str = Dhall.TextLit (Dhall.Chunks [] str)
 
+-- | Merge maps and error on conflicts
+mergeNoConflicts :: (Ord k, Eq a, Show a, Show k) => Data.Map.Map k a -> Data.Map.Map k a -> Data.Map.Map k a
+mergeNoConflicts = Data.Map.unionWithKey
+                   (\key left right ->
+                     if   left == right 
+                     then left 
+                     else error ("Cannot merge differing values " ++ show left ++ " and " ++ show right ++ " for key " ++ show key))
+
+{- | Extract the 'ModelName' to be used when splitting a definition.
+
+    This is considered a guess as it does not work with all types. Currently it uses the first word from the description
+    appended to the largest prefix before the last @.@ of the parent.
+-}
+guessModelNameForSplit :: ModelHierarchy -> Definition -> Maybe ModelName
```

Comment:
I'm a bit concerned about guessing the model name from the first word of the description, since I'm still new to this domain and I don't know how much we can rely on users following this convention.  The main thing I'm trying to avoid is having to help users with debugging their CRD if things go wrong.  The other reason I bring this up is because it seems like we don't need to worry about the splits being particularly ergonomic since they only need to be specified once.

One intermediate alternative is perhaps to allow the user to omit the prefix from the model name, which would default to using the parent's prefix.  In other words, the user could specify:

```
com.example.v1.Certificate~spec=CertificateSpec
```

Reply:
Here is my sample of CRDs that led me to think this was a good idea:
| CRD | Follows Convention |
| --- | --- |
| kubevirt | ✓ |
| cert-manager | ✗ |
| calico | ✓ |
| contour | ✓ |
| rook | ✗  |
| non public from my team | ✓ |

I would have to think on making the parents prefix optional. The one use case I wanted to support in the future is:
```
(com.example.ResourceSpec).podSelector=io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector
```
(Right now the only thing preventing that is `LabelSelector` has imports that a naiive `==` on `Dhall.Expr` cant see through.)

Reply:
Alright, then we can keep it the way it is